site/src/api/typesGenerated.ts

```diff
@@ -755,7 +755,7 @@ export type Entitlement = "entitled" | "grace_period" | "not_entitled";
 
 // From codersdk/deployment.go
 export interface Entitlements {
-	readonly features: Record<FeatureName, Feature>;
+	readonly features: Record<FeatureName, Feature> | null;
```

Comment:
@Emyrk is it true? I think features field is always being returned right?

Reply:
This is technically correct. We should always return something, but it is possible to return a `null` from the golang side of things.

Reply:
Sure, but we should be able to always represent data accurate even it being possible to return null. I'm start to think we should maybe use a tool like this https://github.com/openapi-ts/openapi-typescript since we already generates the openapi spec. Wdyt?

Reply:
We can revert this change, and get the type back to the way it was. I decided to make things closer to the truth because of this error we hit in production.

https://github.com/coder/coder/pull/17430

We assumed the map was never `null`, but there was an edge case where it was, and things broke. If we went to the Open API spec, we have the same issue.

----

An idea we could do is make a new `Record<FeatureName, Feature>` that unmarshals `null` into `{}`. We could do that on the backend or the frontend.

Would that be difficult you think on the frontend? To interpret `null` as an empty Record?